apps/admin-portal/src/components/applications/application-edit.tsx

```diff
@@ -0,0 +1,130 @@
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import React, { FunctionComponent, useEffect, useState } from "react";
+import { Button, Divider, Form, Grid, Icon, Tab } from "semantic-ui-react";
+import { getApplicationDetails } from "../../api/application";
+import { history } from "../../helpers";
+import { ApplicationInterface } from "../../models";
+import { GeneralDetailsApplication } from "./general-details-application";
+import { ApplicationSettings } from "./settings-application";
+
+/**
+ * Application edit component.
+ *
+ * @return {JSX.Element}
+ */
+export const EditApplication: FunctionComponent<{}> = (props): JSX.Element => {
+
+    const [application, setApplication] = useState<ApplicationInterface>();
+
+    const setBasic = (basic: ApplicationInterface) => {
+        setApplication(basic);
+    };
+
+    const getAppID = (): string => {
+        const path = history.location.pathname.split("/");
+        const appName = path[path.length - 1];
+        return appName;
+    };
+
+    const ApplicationDetails = (id: string) => {
+        getApplicationDetails(id)
+            .then((response) => {
+                setBasic(response);
+            })
+            .catch((error) => {
+                // TODO add to notifications
+            });
+    };
+
+    useEffect(() => {
+        ApplicationDetails(getAppID());
+    }, []);
+
+    const navigate = (): void => {
+        history.push("/applications");
+    };
+
+    const panes = () => ([
+        {
+            menuItem: "General",
+            render: () => (
+                <Tab.Pane attached={ false }>
+                    <GeneralDetailsApplication
+                        appId={ application.id }
+                        name={ application.name }
+                        description={ application.description }
+                        imageUrl={ application.imageUrl }
+                        accessUrl={ application.accessUrl }
+                    />
+                </Tab.Pane>
+            ),
+        },
+        {
+            menuItem: "Settings",
+            render: () => (
+                <Tab.Pane attached={ false }>{
+                    (
+                        <ApplicationSettings
+                            appId={ application.id }
+                            advancedConfigurations={ application.advancedConfigurations }
+                        />
+                    )
+                }
+                </Tab.Pane>
+            ),
+        },
+        {
+            menuItem: "SignOnMethods",
+            render: () => <Tab.Pane attached={ false }>SignOnMethod</Tab.Pane>,
+        },
+    ]);
+
+    return (
+        <>
+            { application &&
```

Comment:
I think we are missing ? mark and some other condition here.

Reply:
This is not a conditional (ternary) operator, it is more like a circuit breaker if "application" exists second expression will execute.